src/clientdlg.cpp

```diff
@@ -33,18 +33,20 @@ CClientDlg::CClientDlg ( CClient*         pNCliP,
                          const bool       bShowAnalyzerConsole,
                          const bool       bMuteStream,
                          const bool       bNEnableIPv6,
+                         const bool       bNEnableAccessiblePushButtonUi,
                          QWidget*         parent ) :
     CBaseDlg ( parent, Qt::Window ), // use Qt::Window to get min/max window buttons
     pClient ( pNCliP ),
     pSettings ( pNSetP ),
     bConnectDlgWasShown ( false ),
     bDetectFeedback ( false ),
     bEnableIPv6 ( bNEnableIPv6 ),
+    bEnableAccessiblePushButtonUi ( bNEnableAccessiblePushButtonUi ),
     eLastRecorderState ( RS_UNDEFINED ), // for SetMixerBoardDeco
     eLastDesign ( GD_ORIGINAL ),         //          "
     ClientSettingsDlg ( pNCliP, pNSetP, parent ),
     ChatDlg ( parent ),
-    ConnectDlg ( pNSetP, bNewShowComplRegConnList, parent ),
+    ConnectDlg ( pNSetP, bNewShowComplRegConnList, bNEnableIPv6, bNEnableAccessiblePushButtonUi, parent ),
```

Comment:
the ipv6 toggle wasn't passed correctly.

Reply:
It's another indicator of the overall design problem.

Passing parameters is just the wrong way to do it.  This should be signals on CClient with slots wired inside CConnectDlg to get notifications of updates to the values, and getters on CClient to supply the current state.  Only CClient should provide this information.  Even things like the IPv6 toggle _should_ be controllable through the UI -- so you'd need the CConnectDlg to track changes in that state.  CClientDlg would then get passed a thread-safe reference to CClient.

(All for r4.0.0, naturally.)